Paymetheus/ViewModels/PurchaseTicketsViewModel.cs

```diff
@@ -0,0 +1,404 @@
+                }
+
+                if (_testPoolFees < 1.0)
+                {
+                    _poolFees = 0.0;
+                    throw new ArgumentException("Bad pool fees; pool fees must be greater than 0.00%");
+                }
+
+                _poolFees = value;
+                _settingsAreValid();
+            }
+        }
+
+        private long _number = 0;
+        public long Number
+        {
+            get { return _number; }
+            set
+            {
+                if (value < 0)
+                {
+                    _number = 0;
+                    throw new ArgumentException("Negative number of tickets passed");
+                }
+
+                if (_selectedAccount == null)
+                {
+                    _number = 0;
+                    throw new ArgumentException("No account selected");
+                }
+
+                if (_selectedAccount.Balances.SpendableBalance <= 0) {
+                    _number = 0;
+                    throw new ArgumentException("Empty account");
+                }
+
+                if ((Amount)(_stakeDifficultyProperties.Price * value) > _selectedAccount.Balances.SpendableBalance)
+                {
+                    _number = 0;
+                    string errorString = "Not enough funds; have " +
+                        _selectedAccount.Balances.SpendableBalance.ToString() + " want " +
+                        ((Amount)(_stakeDifficultyProperties.Price * value)).ToString();
+                    throw new ArgumentException(errorString);
+                }
+
+                _number = value;
+                _settingsAreValid();
+            }
+        }
+
+        // The default expiry is 16.
+        private int _expiry = 16;
+        private int _minExpiry = 2;
+        public int Expiry
+        {
+            get { return _expiry; }
+            set
+            {
+                if (value <= _minExpiry)
+                {
+                    _expiry = 0;
+                    throw new ArgumentException("Expiry must be a minimum of 2 blocks");
+                }
+
+                _expiry = value;
+                _settingsAreValid();
+            }
+        }
+
+        // TODO Declare this as a global somewhere?
+        private Amount _minFeeAmount = Denomination.Decred.AmountFromDouble(0.01000000);
+        private Amount _maxFeeAmount = Denomination.Decred.AmountFromDouble(0.99999999);
+
+        private Amount _splitFee = 0;
+        public double SplitFee
+        {
+            get { return Denomination.Decred.DoubleFromAmount(_splitFee); }
+            set
+            {
+                var _testAmount = Denomination.Decred.AmountFromDouble(value);
+
+                if (_testAmount < _minFeeAmount)
+                {
+                    _splitFee = 0;
+                    throw new ArgumentException("Too small fee passed (must be >= 0.01000000 DCR/KB)");
```

Comment:
what's the point of declaring a const if we're just going to hardcode the value in a string elsewhere?

`throw new ArgumentException($"Fee/kB is too small ({parsedFee} < {(Amount)MinFeePerKb}");`